### PR TITLE
 small modification to prevent elixir 1.15 deprecations

### PR DESCRIPTION
**Pull request by JoeZ99**
these are just so elixir 1.15 doesn't throw warnings ...

### CODE REVIEW

File: lib/codepagex/mappings.ex [hunk 1]
```diff
@@ -108,13 +108,8 @@ defmodule Codepagex.Mappings.Helpers do
     end
   end
 
-  defp name_matches?(name, filter) do
-    if Regex.regex?(filter) do
```
Comment (JoeZ99) [inline lib/codepagex/mappings.ex:112]: `Regex.regex?/1` is deprecated in elixir 1.14

File: lib/codepagex/mappings.ex [hunk 2]
```diff
@@ -177,7 +172,7 @@ defmodule Codepagex.Mappings do
 
   @filtered_names_files Helpers.filter_to_selected_encodings(
                           @all_names_files,
-                          Application.get_env(:codepagex, :encodings, @default_mapping_filter),
```
Comment (JoeZ99) [inline lib/codepagex/mappings.ex:180]: they encourage `compile_env` instead of `get_env`

**Comment by tallakt** [comment]
Thanks!

**Comment by JoeZ99** [comment]
> Thanks!

You're welcome! I really need a new version (maybe `0.1.7` ??) with this be available, is there some timetable for that???

**Comment by tallakt** [comment]
i can push it in the morning

**Comment by tallakt** [comment]
I was not able to push, so many error due to old dependencies. I will need more time.

**Comment by JoeZ99** [comment]
> I was not able to push, so many error due to old dependencies. I will need more time.

sorry for being such a pushy, and I really appreciate the struggle you're going through. Take your time , though, I will happily adapt to that. I just upgraded some project to elixir 1.15 and your library was in `mix.exs`, that's all :smile:

**Pull request merged**